wpilib/wpilib/analogaccelerometer.py

```diff
@@ -35,19 +34,27 @@ def __init__(self, channel):
         :param channel: port index or an already initialized :class:`.AnalogInput`
         """
         super().__init__()
-        if not hasattr(channel, "getAverageVoltage"):
-            channel = AnalogInput(channel)
-        self.analogChannel = channel
+        if not hasattr(channel, "getAverageVoltage"): # If 'channel' is an integer
+            self.analogChannel = AnalogInput(channel)
+            self.allocatedChannel = True
+            self.addChild(self.analogChannel)
+        else:
+            self.allocatedChannel = False
+            self.analogChannel = channel
         self.voltsPerG = 1.0
         self.zeroGVoltage = 2.5
         self.pidSource = self.PIDSourceType.kDisplacement
         hal.report(hal.UsageReporting.kResourceType_Accelerometer,
                       self.analogChannel.getChannel())
-        LiveWindow.addSensorChannel("Accelerometer",
+        self.setName("Accelerometer",
                                     self.analogChannel.getChannel(), self)
```

Comment:
Something seems a little off.

I guess this means we have absolutely zero coverage of AnalogAccelerometer. Could you add a few basic tests?